catalog/urls.py

```diff
@@ -5,6 +5,7 @@
 app_name = "catalog"
 
 urlpatterns = [
-    path("home/", views.home, name="home"),
     path("contacts/", views.contacts, name="contacts"),
+    path("product_detail/<int:product_id>", views.product_detail, name="product_detail"),
```

Comment:
В конце нужен слеш, а вообще правильнее назвать эндпоинт так:
```suggestion
    path("products/<int:product_id>/detail/", views.product_detail, name="product_detail"),
```